cli/src/main/java/org/jboss/pnc/bacon/cli/App.java

```diff
@@ -17,156 +17,189 @@
+
+            // Add more loggers that you want to switch to DEBUG here
+            ObjectHelper.setLoggingLevel("org.jboss.pnc.client", Level.DEBUG);
+
+            log.debug("Log level set to DEBUG");
+        }
+    }
+
+    @Option(names = "--profile", defaultValue = "default", description = "PNC Configuration profile", scope = INHERIT)
+    public void setProfile(String profile) {
+        this.profile = profile;
+    }
+
+    /**
+     * Set the path to config file if the configPath flag or environment variable is set
+     */
+    @Option(names = "-p", description = "Path to PNC configuration folder", scope = INHERIT)
+    public void setConfigurationFileLocation(String configPath) {
+        if (configPath != null) {
+            setConfigLocation(configPath, "flag");
+        }
+    }
+
+    @Option(names = { "-h", "--help" }, usageHelp = true, description = "display this help message", scope = INHERIT)
+    boolean usageHelpRequested;
 
-    public int run(String[] args) throws Exception {
+    @Option(names = { "-V", "--version" }, versionHelp = true, description = "display version info")
+    boolean versionInfoRequested;
 
-        CommandRegistry registry;
+    public int run(String[] args) {
+        CommandLine commandLine = new CommandLine(this);
+        commandLine.setUsageHelpAutoWidth(true);
 
         if (args.length == 0) {
-            registry = AeshCommandRegistryBuilder.builder()
-                    .command(Pnc.class)
-                    .command(Da.class)
-                    .command(Pig.class)
-                    .command(CommandBuilder.builder().name("exit").command(commandInvocation -> {
-                        commandInvocation.stop();
-                        return CommandResult.SUCCESS;
-                    }).create())
-                    .create();
-            SettingsBuilder<CommandInvocation, ConverterInvocation, CompleterInvocation, ValidatorInvocation, OptionActivator, CommandActivator> builder = SettingsBuilder
-                    .builder()
-                    .logging(true)
-                    .enableAlias(false)
-                    .enableExport(false)
-                    .enableMan(true)
-                    .enableSearchInPaging(true)
-                    .readInputrc(false)
-                    .commandRegistry(registry);
-
-            ReadlineConsole console = new ReadlineConsole(builder.build());
-            console.setPrompt(
-                    new Prompt(
-                            new TerminalString(
-                                    "[bacon@console]$ ",
-                                    new TerminalColor(Color.DEFAULT, Color.DEFAULT, Color.Intensity.BRIGHT))));
-
-            console.start();
-
-            return 0;
-        } else {
-            registry = AeshCommandRegistryBuilder.builder().command(this.getClass()).create();
+            // From https://github.com/remkop/picocli/wiki/JLine-3-Examples
+            AnsiConsole.systemInstall();
```

Comment:
I used this line in build finder, but I couldn't figure out how to guarantee how `AnsiConsole.systemUninstall()` would be called, so I put the install in a `static` block and uninstall in a shutdown handler. You _must_ call uninstall once for every install or there is some resource leak.

But, now I think this is actually only used on old Windows versions (< 10), so it may be able to be left out entirely. You can even see in the picocli code logic like

```
if (isWindows && !Ansi.AUTO.enabled()) {
    AnsiConsole.systemInstall();
}
```

Reply:
I guess I could put the uninstall in the EndOfFileException block (when ctrl-d) is called which exits the program. However I can't see where your code is sourced from - I used https://github.com/remkop/picocli/tree/master/picocli-shell-jline3

Reply:
I may have overexaggerated the need for it---it looks like uninstall just resets the I/O streams, but it still looks like a bug in that jline3 example.

The one I was talking about is from here: https://github.com/remkop/picocli/blob/34a7d3ab72563678303dd02e683f1ad599732c49/src/test/java/picocli/HelpAnsiTest.java#L118 and Jansi's own examples: https://github.com/fusesource/jansi/blob/146109610a85892efdad4f50a1a974d9d6ac21ee/example/src/main/java/org/fusesource/jansi/Main.java#L10

Reply:
I have added an uninstall in the latest version for when the user quits the console. Perhaps you could enter an issue in picocli jline3 - the author is very responsive.

Reply:
I'll file an issue against picocli. Maybe he knows whether it is necessary, too.

Reply:
He replied to https://github.com/remkop/picocli/issues/1165 and added the uninstall to the docs. It also seems that this could be made conditional or even removed. See what you think after you read the discussion on that issue.